storagenode/notifications/notifications.go

```diff
@@ -0,0 +1,73 @@
+// Copyright (C) 2019 Storj Labs, Inc.
+// See LICENSE for copying information.
+
+package notifications
+
+import (
+	"context"
+	"time"
+
+	"github.com/skyrings/skyring-common/tools/uuid"
+
+	"storj.io/storj/pkg/storj"
+)
+
+// DB tells how application works with notifications database.
+//
+// architecture: Database
+type DB interface {
+	Insert(ctx context.Context, notification NewNotification) (Notification, error)
+	List(ctx context.Context, cursor NotificationCursor) (NotificationPage, error)
+	Read(ctx context.Context, notificationID uuid.UUID) error
+	ReadAll(ctx context.Context) error
+}
+
+// NotificationType is a numeric value of specific notification type.
+type NotificationType int
+
+const (
+	// NotificationTypeCustom is a common notification type which doesn't describe node's core functionality.
+	NotificationTypeCustom NotificationType = 0
```

Comment:
I feel like this Type will be misused in the future.
We should make something along the lines of Information..

Reply:
now we don't know all custom notification types. Let's wait for all the info about it. I'll add TODO for this. Are you ok with that?

Reply:
custom notification can have any topic, for example it can be notification about new blogpost, New Year sale, payouts made etc.